### ISSUE TITLE
Support functions defines as const

### ISSUE THREAD

**Issue by msmfsd** [esdoc/esdoc#318]
Functions declared as const's get documented as variables and not functions eg:

```
const myFunct = (i) => { return i + 1; }
```

Is there a way to display them as functions in the docs?

**Comment by doemijnina** [comment]
I am actually changing my function declarations back to function expressions because of this issue. Would really like to know if there is a better way to handle this.

**Comment by Freezystem** [comment]
I have the same problem in my code. I'm forced to used normal `function` declaration in my code or my fat arrow functions declared with `const` are shown as standard variables in the doc and all param descriptions aren't available.
I've also tried to force the type using `@type {function}` but the parser seems to bypass the instruction.
IMO the `@type {function}` is the best compromise to deal with this case.

**Comment by Freezystem** [comment]
@h13i32maru, @typhonrt sorry to bother but have you any useful tips on that point ?
Is that a constraint coming from babylon parser or is it possible to mitigate this issue ?
Thanks in advance for your time.

**Comment by msmfsd** [comment]
bump..

**Comment by typhonrt** [comment]
This is indeed a current limitation. Anything is possible to arrange as far as marking an arrow function assigned to a variable as a function instead of a variable regardless of AST parser being used. It can get a bit wonky though. Is this only supported for `const` assignments? Otherwise a state machine of sorts needs to be created when parsing a file / module to detect when an arrow function assigned to a variable is overridden:

```
let myFunction = () => { ... }
myFunction = 1;
```

Likewise with a class in a file / module assigning an imported function, an arrow function or any other function to the class prototype should be handled. IE:

```
import aFunction from './aFunction';

export default class MyClass { ... }

MyClass.prototype.myFunction = () => { ... }
MyClass.prototype.myFunction2 = function() { ... }

MyClass.prototype.aFunction = aFunction;

// However what about the non-likely situation...
MyClass.prototype.myFunction = 1;
```

All of the above require a state machine at the module / file level. And to support an imported function (`afunction`) would require state tracking across files for imports / exports. 

Basically there is no problem technically in solving this category of issues. It is significant enough though that attempting a PR that may or may not be accepted is time consuming. @h13i32maru will have to acknowledge this issue and consider it, but given his time limitations towards ESDoc in general I'd expect such a proposal would be deferred.

**Comment by Freezystem** [comment]
Thanks for your time @typhonrt. 
I think I get your point so i'd say that recognizing a variable as a function should only be supported for the `const` keyword. Cause once it's set, it can't be changed afterwards event within the module, avoiding the use of a state machine.

In general way, is `export const myFunc = () => {...};` a good coding pattern or is it more careful to use `export function myFunc () {...}` regarding parser compatibilities and standards ?

**Comment by typhonrt** [comment]
I use the latter style `export function myFunc () {...}` as this is explicitly picked up by ESDoc currently. 

If and when handling arrow functions assigned to const variables is added to ESDoc then the documentation between both styles you posted should be equivalent. Personally I think the latter option is also a bit more readable. 

It's hard to say though if and when support of arrow functions will be supported as functions in the docs. The const assignment limitation is easier to support, but needs to be clearly stated in ESDoc documentation & tutorials as it is still kind of a piecemeal partial solution. ESDoc current doesn't do introspection so it relies on the ES module AST / spec for explicit doc identifiers. If I had any say I think going to full introspection route while more difficult is the best overall solution.

**Comment by garetmckinley** [comment]
Currently refactoring my code base because I developed a react app without knowing esdoc had no support for arrow functions 😭 

Subscribing to this issue in case theres any progress 👍

**Comment by renaudl** [comment]
With all the fuzz arround react and redux using exclusively const arrow function, esdoc should be able to support it. This does not make sens to refactor your code because of a code generation issue really. If ESDOC could do this, it would be fantastic.

**Comment by h13i32maru** [comment]
ESDoc v0.5.0 supports `arrow function exporting`. Please check it.

This code is ok.
```js
/**
 * this is arrow function.
 * @param {number} a - this is a.
 * @param {number} b - this is b.
 */
export default (a, b)=>{}

const f = (x, y)=>{}
/**
 * this is f.
 */
export {f}
```

reopen if is problem.

**Comment by Freezystem** [comment]
Hi! Big thanks for your time and work. I've just tested to re-render my documentation using arrow function and it worked like a charm ! 💯

**Comment by nirajbhutada** [comment]
I have installed esDoc 0.5.0, but still its not working dude 👎
